test/functional/update/update-iterative-download-fail.bats

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bats
+
+load "../testlib"
+
+test_setup() {
+
+	create_test_environment "$TEST_NAME"
+	create_bundle -L -n test-bundle -f /foo,/bar,/bat/baz "$TEST_NAME"
+	create_version "$TEST_NAME" 20 10
+	update_bundle -i "$TEST_NAME" test-bundle --update /foo
+	sudo rm -rf "$WEBDIR"/20/Manifest.test-bundle.I.10
+	sudo rm -rf "$WEBDIR"/20/Manifest.test-bundle.I.10.tar
+
+}
+
+@test "update falls back to full manifests when the iterative fails to download" {
+
+	run sudo sh -c "$SWUPD update $SWUPD_OPTS_KEEPCACHE"
+
+	assert_status_is 0
+
+	# verify target files
+	assert_file_exists "$TARGETDIR"/foo
+	assert_file_exists "$TARGETDIR"/bar
+	assert_file_exists "$TARGETDIR"/bat/baz
+
+	# since the iterative manifest was deleted the full manifest should
+	# have been downloaded
+	assert_file_exists "$STATEDIR"/20/Manifest.test-bundle
```

Comment:
Should you also assert that the iterative manifest is in the MoM so we know that swupd will have tried to use an iterative manifest and failed?

Reply:
It's a single line of bash to make sure we don't regress and fail to add the record to the MoM. It ensures we are actually testing what we expect to be testing.

Reply:
ok, I'll add that to the tests